crates/proto/src/xfer/dnssec_dns_handle/nsec3_validation.rs

```diff
@@ -1,3 +1,72 @@
+//! Throughout this module several NSEC3 specific terms are used.
+//!
+//! "closest_encloser" - a name that is a parent / grandparent /
+//!     great-grandparent, etc. of a `query_name`, that DOES exist.
+//!
+//! "next_closer" - a name that is one level deeper in the hierarchy from
+//!     `closest_encloser`, that DOES NOT exist in the zone.
+//!
+//! "wildcard of closest encloser" - a name in a form of `*.closest_encloser`.
+//!     If it exists then `query_name` would be serviced by the wildcard.
+//!
+//! "covering NSEC3 record" - NSEC3 record name has a hash (`hashed_owner_name`),
+//!     and inside the record data there's `next_hashed_owner_name`.
+//!     If the hash of `query_name` fits between the two hashes then the record
+//!     "covers" `query_name`
+//!
+//! "own NSEC3 record" - exists for *existing* names only.
```

Comment:
I like "own" but "matching" is the term used in the RFC. Not a big deal for me anyway.